cmd/genesis/genesis.go

```diff
@@ -0,0 +1,139 @@
+				PrefundedAccounts:             parsePrefundedAccounts(viper.Get("prefundedAccounts")),
+				ConsensusEngine:               viper.GetString("genesis.consensusEngine"),
+				SmartContractsOwner:           viper.GetString("genesis.smartContractsOwner"),
+				ExtraData:                     viper.GetString("genesis.extraData"),
+			}
+
+			fileName := "genesis.json"
+			if viper.GetString("genesis.fileName") != "" {
+				fileName = viper.GetString("genesis.fileName")
+			}
+
+			file, err := os.Create(fileName)
+			if err != nil {
+				fmt.Printf("Error generating file: %s", err)
+				os.Exit(1)
+			}
+
+			handler := generateGenesisFileCommandHandler{w: file}
+			err = handler.handle(command)
+			if err != nil {
+				fmt.Printf("Error generating file: %s", err)
+				os.Exit(1)
+			}
+
+			fmt.Println("Genesis file generated.")
+		},
+	}
+
+	cmd.Flags().StringVarP(&FileConfig, "config", "c", "", "Use to load configuration from config file.")
+	cmd.Flags().StringP("network", "n", "", "The network to use, test or main")
+	viper.BindPFlag("genesis.network", cmd.Flags().Lookup("network"))
+	cmd.Flags().StringP("maxNumValidators", "v", "", "The maximum num of validators.")
+	viper.BindPFlag("genesis.maxNumValidators", cmd.Flags().Lookup("maxNumValidators"))
+	cmd.Flags().StringP("unbondingPeriod", "p", "", "The unbonding period in days.")
+	viper.BindPFlag("genesis.unbondingPeriod", cmd.Flags().Lookup("unbondingPeriod"))
+	cmd.Flags().StringP("walletAddressGenesisValidator", "g", "", "The wallet address of the genesis validator.")
+	viper.BindPFlag("genesis.walletAddressGenesisValidator", cmd.Flags().Lookup("walletAddressGenesisValidator"))
+	cmd.Flags().StringP("consensusEngine", "e", "", "The consensus engine, right now only supports tendermint")
+	viper.BindPFlag("genesis.consensusEngine", cmd.Flags().Lookup("consensusEngine"))
+	cmd.Flags().StringP("smartContractsOwner", "s", "", "The address of the smart contracts owner.")
+	viper.BindPFlag("genesis.smartContractsOwner", cmd.Flags().Lookup("smartContractsOwner"))
+	cmd.Flags().StringP("extraData", "d", "", "Extra data")
+	viper.BindPFlag("genesis.extraData", cmd.Flags().Lookup("extraData"))
+	cmd.Flags().StringP("prefundedAccounts", "a", "", "The prefunded accounts in format 0x212121:12,0x212121:14")
+	viper.BindPFlag("prefundedAccounts", cmd.Flags().Lookup("prefundedAccounts"))
+	cmd.Flags().StringP("fileName", "o", "", "The output filename (default:genesis.json).")
+	viper.BindPFlag("genesis.fileName", cmd.Flags().Lookup("fileName"))
+}
+
+func loadFromFileConfigIfAvailable() {
+	if FileConfig != "" {
+		viper.SetConfigFile(FileConfig)
+
+		err := viper.ReadInConfig()
+		if err != nil {
+			panic(fmt.Errorf("Fatal error config file: %s \n", err))
+		}
+	}
+}
+
+func main() {
+	if err := cmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func parsePrefundedAccounts(accounts interface{}) []genesis.PrefundedAccount {
```

Comment:
This is needed since data can come from 2 types, string or slice depending on where it is loaded (config or command line). I preferred to have this encapsulated in this function instead of having an if on the callee function. It is not the best, but not really really the worst. If you still want to fix this let me know! :D

Reply:
I'd also prefer to either be specific with the data we're sending (if it's a string, use a separate func than the slice), or use always slices, even if there's only one element. Is that possible?

Reply:
if it overcomplicates things, then I'm fine with this. @jmatosp?